contracts/JBveBanny.sol

```diff
@@ -239,34 +216,52 @@ contract JBveBanny is ERC721Votes, ERC721Enumerable, Ownable, ReentrancyGuard, J
   {
     if (_useJbToken) {
       // If a token wasn't set when this contract was deployed but is set now, set it.
-      if (token == IJBToken(address(0)) && tokenStore.tokenOf(projectId) != IJBToken(address(0))) {
-        token = tokenStore.tokenOf(projectId);
+      if (token == address(0) && tokenStore.tokenOf(projectId) != IJBToken(address(0))) {
+        token = address(tokenStore.tokenOf(projectId));
         // The project's token must not have changed since this token was originally set.
-      } else if (tokenStore.tokenOf(projectId) != token) revert TOKEN_MISMATCH();
+      } else if (address(tokenStore.tokenOf(projectId)) != token) revert TOKEN_MISMATCH();
     }
 
     // Duration must match.
     if (!_isLockDurationAcceptable(_duration)) revert JBErrors.INVALID_LOCK_DURATION();
 
     // Make sure the token balance of the account is enough to lock the specified _count of tokens.
-    if (_useJbToken && token.balanceOf(_account, projectId) < _count)
+    if (_useJbToken && IJBToken(token).balanceOf(_account, projectId) < _count)
```

Comment:
Could we not remove these and instead just perform the transfers? Those will revert if the user does not have enough balance and that would save us a call

Reply:
yeah I mean it's just a precautionary check but yeah we can remove